packages/adapter-components/src/elements/swagger/type_elements/type_config_override.ts

```diff
@@ -113,44 +114,18 @@ export const fixTypes = (
     return type
   }
 
-  const getTypeTransformationConfig = (typeName: string): TransformationConfig => (
-    getConfigWithDefault(
-      typeConfig[typeName]?.transformation,
-      typeDefaultConfig.transformation,
-    )
-  )
-
   // fix field types
-  Object.keys(typeConfig)
-    .filter(typeName => getTypeTransformationConfig(typeName).fieldTypeOverrides !== undefined)
-    .forEach(typeName => {
-      const type = definedTypes[typeName]
-      if (type === undefined) {
-        log.warn('type %s not found, cannot override its field types', typeName)
-        return
-      }
-      const fieldTypeOverrides = getTypeTransformationConfig(
-        typeName
-      ).fieldTypeOverrides as FieldTypeOverrideType[]
-      fieldTypeOverrides.forEach(({ fieldName, fieldType }) => {
-        const field = type.fields[fieldName]
-        const newFieldType = toTypeWithContainers(fieldType)
-        if (field === undefined) {
-          log.debug('Creating field type for %s.%s with type %s', typeName, fieldName, newFieldType.elemID.name)
-          type.fields[fieldName] = new Field(type, fieldName, newFieldType)
-        } else {
-          log.debug('Modifying field type for %s.%s from %s to %s', typeName, fieldName, field.refType.elemID.name, newFieldType.elemID.name)
-          field.refType = createRefToElmWithValue(newFieldType)
-        }
-      })
-    })
+  fixFieldTypes(definedTypes, typeConfig, typeDefaultConfig, toTypeWithContainers)
 
   // hide env-specific fields
   Object.entries(definedTypes)
-    .filter(([typeName]) => getTypeTransformationConfig(typeName).fieldsToHide !== undefined)
+    .filter(([typeName]) =>
+      getTypeTransformationConfig(typeName, typeConfig, typeDefaultConfig)
+        .fieldsToHide !== undefined)
```

Comment:
readability nit:
```suggestion
      getTypeTransformationConfig(
        typeName, typeConfig, typeDefaultConfig,
      ).fieldsToHide !== undefined)
```
(and same below)